mesonbuild/backend/vs2010backend.py

```diff
@@ -870,6 +870,46 @@ def _prettyprint_vcxproj_xml(self, tree, ofname):
             of.write(doc.toprettyxml())
         replace_if_different(ofname, ofname_tmp)
 
+    def gen_vcxproj_filters(self, target, ofname):
+        # Generate pitchfork of filters based on directory structure.
+        root = ET.Element('Project', {'ToolsVersion': '4.0',
+                                      'xmlns': 'http://schemas.microsoft.com/developer/msbuild/2003'})
+        filter_folders = ET.SubElement(root, 'ItemGroup')
+        filter_items = ET.SubElement(root, 'ItemGroup')
+        mlog.debug(f'Generating vcxproj filters {target.name}.')
+        found_folders = []
+        for i in target.sources:
+            path = i.fname.replace('/', '\\')
+            dirname = os.path.dirname(path)
+            while dirname:
+                if dirname and dirname not in found_folders:
+                    found_folders.append(dirname)
+                dirname = os.path.dirname(dirname)
```

Comment:
I think this should be:
```python
while dirname:
    dirname = os.path.dirname(dirname)
    if not dirname or dirname in found_folders:
        break
    found_folders.append(dirname)
```

Because if we ever find a directory already in the path, then all of it's parents should already be there right?
Give `a\b\c\d\<files>` and `a\b\c\e\<files>`, after the first iteration `found_folders` would be `['a', 'b', 'c', 'd']`, so once we'd finished proccessing `e` in the second iteration it would all be matches?

Reply:
Yep, that would be better.